tspng/implantation.py

```diff
@@ -0,0 +1,82 @@
+# import statements
+import os
+import json
+
+from PIL import Image
+from PIL.PngImagePlugin import PngInfo
+from tspng import MIME_TYPE
+from typing import Dict
+
+
+def _implant_data(data: str, image: str, mime_type: str = MIME_TYPE, dst: str = None):
+    # open image
+    target_im = Image.open(image)
+    if target_im.format != "PNG":
```

Comment:
This check for a PNG can be removed. The source image file could be any Pillow image format, PNG, TIFF, etc. and implanting will convert it to a TS png file (PNG). There is no restriction for the source image being PNG.